patches/server/0739-Implemented-OP-level-API.patch

```diff
@@ -0,0 +1,256 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: MCMDEV <john-m.1@gmx.de>
+Date: Thu, 5 Aug 2021 17:00:10 +0200
+Subject: [PATCH] Implemented OP level API
+
+
+diff --git a/src/main/java/net/minecraft/server/dedicated/DedicatedPlayerList.java b/src/main/java/net/minecraft/server/dedicated/DedicatedPlayerList.java
+index 918f5221e94cbc867349c69c83563e225d2fef1d..734e5305d2a059738df20ea9a72dad6ca6c5b810 100644
+--- a/src/main/java/net/minecraft/server/dedicated/DedicatedPlayerList.java
++++ b/src/main/java/net/minecraft/server/dedicated/DedicatedPlayerList.java
+@@ -52,6 +52,17 @@ public class DedicatedPlayerList extends PlayerList {
+         this.saveOps();
+     }
+ 
++    // Paper start - OP level API
++    public int getOperatorLevel(GameProfile profile)    {
++        return this.getServer().getProfilePermissions(profile);
++    }
++
++    public void setOperatorLevel(GameProfile profile, int value)    {
++        super.op(profile, value);
++        this.saveOps();
++    }
++    // Paper end
++
+     @Override
+     public void reloadWhiteList() {
+         this.loadWhiteList();
+diff --git a/src/main/java/net/minecraft/server/players/PlayerList.java b/src/main/java/net/minecraft/server/players/PlayerList.java
+index 48045993c8ad4b014cf4a67f7c4db42e014d1c81..03934348739d31cd41bf74992e59f344adf86669 100644
+--- a/src/main/java/net/minecraft/server/players/PlayerList.java
++++ b/src/main/java/net/minecraft/server/players/PlayerList.java
+@@ -1124,15 +1124,20 @@ public abstract class PlayerList {
+         return this.ipBans;
+     }
+ 
++    // Paper start - OP level API
+     public void op(GameProfile profile) {
+-        this.ops.add(new ServerOpListEntry(profile, this.server.getOperatorUserPermissionLevel(), this.ops.canBypassPlayerLimit(profile)));
++        this.op(profile, this.server.getOperatorUserPermissionLevel());
++    }
++
++    public void op(GameProfile profile, int permissionLevel)    {
```

Comment:
Thanks for the implementation! Already less of a diff, you can now actually adjust your `//Paper start` and `// Paper end` comments respectively (the github diff view shows nicely where your changes now start/end).

Also the new method also has 4 spaces between the method signature and the opening `{` 😅